v0/destinations/kustomer/transform.js

```diff
@@ -0,0 +1,225 @@
+// type of events.
+const constructKustomerPayload = (message, category, email) => {
+  const kustomerPayload = constructPayload(
+    message,
+    MAPPING_CONFIG[category.name]
+  );
+
+  const firstName = getFieldValueFromMessage(message, "firstName");
+  const lastName = getFieldValueFromMessage(message, "lastName");
+  const phone = getFieldValueFromMessage(message, "phone");
+  const url = getFieldValueFromMessage(message, "website");
+
+  if (!get(kustomerPayload, "name") && firstName && lastName) {
+    kustomerPayload.name = `${firstName} ${lastName}`;
+  }
+
+  if (email) {
+    kustomerPayload.emails = kustomerPayload.emails
+      ? kustomerPayload.emails
+      : [{ type: "home", email }];
+  }
+
+  if (phone) {
+    kustomerPayload.phones = kustomerPayload.phones
+      ? kustomerPayload.phones
+      : [{ type: "home", phone }];
+  }
+  if (url) {
+    kustomerPayload.urls = [
+      {
+        url
+      }
+    ];
+  }
+
+  const address = getFieldValueFromMessage(message, "address");
+  if (address) {
+    let addrStr;
+    if (typeof address === "string") {
+      addrStr = address;
+    } else {
+      const { street, city, state, postalCode } = address;
+      addrStr =
+        street || city || state || postalCode
+          ? `${street || ""} ${city || ""} ${state || ""} ${postalCode || ""}`
+          : addrStr;
+    }
+    if (typeof addrStr === "string") {
+      kustomerPayload.locations = [
+        {
+          type: "home",
+          address: addrStr
+        }
+      ];
+    }
+  }
+
+  removeUndefinedAndNullValues(kustomerPayload);
+  return kustomerPayload;
+};
+
+// Main process function responsible for building payload for all
+// type of events.
+const responseBuilderSimple = async (message, category, destination) => {
+  let payload = {};
+  let targetUrl;
+  let storedState = {
+    userExists: false
+  };
+  // In case of identify type of event first extract the anonymousId, userId
+  // and search if any Kustomer is present in destination.
+  // If present update the same kustomer with the given payload.
+  // else create a new kustomer.
+  //-------------------------------------------------------------
+  // Get Kustomer: https://apidocs.kustomer.com/#ff41f372-6144-4c64-9712-662ee5ef1c33
+  // Create Kustomer: https://apidocs.kustomer.com/#07bd1072-4d4b-4875-b526-8369d711e811
+  // Update Kustomer: https://apidocs.kustomer.com/#077d653a-184e-4153-8133-d24b6427c1ae
+  if (message.type.toLowerCase() == EventType.IDENTIFY) {
+    const userEmail = getFieldValueFromMessage(message, "email");
+    const userId = getFieldValueFromMessage(message, "userIdOnly");
+    const anonymousId = get(message, "anonymousId");
+    const externalId = get(message, "context.externalId");
+    if (externalId && externalId.id && externalId.type === "kustomerId") {
+      storedState = {
+        userExists: true,
+        targetUrl: `${BASE_ENDPOINT}/v1/customers/${externalId.id}?replace=false`
+      };
+    }
```

Comment:
We've a method in `util/index.js` for fetching the value of the `externalId` directly. Let's use that.

Reply:
done